cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -457,6 +474,62 @@ class FitMultivariateAnomaly(override val uid: String) extends Estimator[DetectM
 
   protected def prepareUrl: String = getUrl
 
+  override protected def queryForResult(key: Option[String],
+                                        client: CloseableHttpClient,
+                                        location: URI): Option[HTTPResponseData] = {
+    val get = new HttpGet()
+    get.setURI(location)
+    key.foreach(get.setHeader("Ocp-Apim-Subscription-Key", _))
+    get.setHeader("User-Agent", s"synapseml/${BuildInfo.version}${HeaderValues.PlatformInfo}")
+    val resp = convertAndClose(sendWithRetries(client, get, getBackoffs))
+    get.releaseConnection()
+    Some(resp)
+  }
+
+  override protected def handlingFunc(client: CloseableHttpClient,
+                                      request: HTTPRequestData): HTTPResponseData = {
+    val response = HandlingUtils.advanced(getBackoffs: _*)(client, request)
+    if (response.statusLine.statusCode == 201) {
+      val location = new URI(response.headers.filter(_.name == "Location").head.value)
+      val maxTries = getMaxPollingRetries
+      val key = request.headers.find(_.name == "Ocp-Apim-Subscription-Key").map(_.value)
+      val it = (0 to maxTries).toIterator.flatMap { _ =>
+        val resp = queryForResult(key, client, location)
+        val fields = IOUtils.toString(resp.get.entity.get.content, "UTF-8").parseJson.asJsObject.fields
+        val status = if (fields.keySet.contains("modelInfo")) {
```

Comment:
I see that this code is duplicated twice (sorry for not catching this first time around). It might be a good idea to either pull this out into a separate method that parses this, or use spray to parse this so that its a quick 1-liner